index.bs

```diff
@@ -4053,164 +4253,973 @@ To <dfn>get the response data</dfn> given |response|:
    field set to |headers|, the <code>mimeType</code> field set to |mime type|, the
    <code>bytesReceived</code> field set to |bytes received|, the
    <code>headersSize</code> field set to |headers size|, the
-   <code>bodySize</code> field set to |body size|, and the <code>content</code>
-   field set to |content|.
+   <code>bodySize</code> field set to |body size|, <code>content</code>
+   field set to |content|, and the <code>authChallenges</code> field set to
+   |auth challenges| if it's not null, or omitted otherwise.
 
 </div>
 
-### Events ### {#module-network-event}
+#### The network.SetCookieHeader Type #### {#type-network-SetCookieHeader}
 
-#### The network.beforeRequestSent Event #### {#event-network-beforeSendRequest}
-<dl>
-   <dt>Event Type</dt>
-   <dd>
-      <pre class="cddl local-cddl">
-        network.BeforeRequestSent = {
-         method: "network.beforeRequestSent",
-         params: network.BeforeRequestSentParameters
-       }
+[=Remote end definition=]
 
-       network.BeforeRequestSentParameters = {
-         network.BaseParameters,
-         initiator: network.Initiator,
-       }
-      </pre>
-   </dd>
-</dl>
+<pre class="cddl local-cddl remote-cddl">
+network.SetCookieHeader = {
```

Comment:
What is the reason for not treating cookies as a regular header?

Reply:
is it only to help the user with the generation of the cookie header string or are there other reasons?

Reply:
@OrKoN It introduces an asymmetry in the API, since cookies are explicitly separated in the request data. Additionally, translating a cookie with the desired data into a `Set-Cookie` header, properly formatted, with the proper fields is likely to be error-prone for client writers.

Either cookies are a special case everywhere, or they’re a special case nowhere. In the existing representation of request data, we’ve already come to a consensus that the former is the right decision.

Reply:
That means that cookies would be included twice in every request? once in the headers and once in the cookies list, is that right?

Reply:
That depends. Do you mean in requests intercepted by the driver, and reported via events? If so, then yes, you’d see a `Cookie` header, consisting of only name-value pairs, and a list of cookies that contain the additional cookie values (domain, path, expiration, etc.), if you mean in requests and/or responses sent via commands as potentially modified/added/removed/edited by users, then that depends entirely on the parameters passed in by the user in the command to provide the modified request or fulfilled response. Or at least that’s my reading.

Making modifications to cookies in requests and responses is a common enough user action that I’m encouraging of making this part of the API easier to use for users. Forcing the user to treat cookies as “just another header,” especially considering the complexity of formatting the header properly, ans well as the cognitive dissonance with already making cookies “special” in the request interception event data, increases friction for users.

Reply:
Do you have links to issues where this topic has been discussed before? It feels like this we might need a better integration with future cookie manipulation story (https://github.com/w3c/webdriver-bidi/issues/287) as I am not sure that it is a good thing to unconditionally include serialisation of metadata cookies on each request.

Reply:
I'm not sure I can point to a ton of other issues. I can point to the [comment I made in May](https://github.com/w3c/webdriver-bidi/pull/429#discussion_r1203152220) which led to this addition by @jgraham. The inclusion of cookie metadata in the [predates this PR](https://w3c.github.io/webdriver-bidi/#type-network-RequestData), which has been part of the spec draft for months.

Reply:
I don't know what a "metadata cookie" is, but I don't think this affects the future cookie manipulation story much. 

The key problem we have with general access to the cookie store (or other storage) is that we don't know in general what the keys to the store look like in each browser. For requests and responses that's not relevant because it's just providing the values that are actually in the request/response in a slightly easier to use form. Indeed, that part was already in network request logging, and matches what CDP does.

All that's new here is the ability to also _modify_ cookies using the easier to use form, rather than writing the header values directly. Really the only complexity is that there are now multiple ways to set the same header, and you need to reconcile them. But of course a client can also choose not to use this feature at all.

Reply:
>  matches what CDP does

I guess that would be `associatedCookies` on the requestWillBeSentExtraInfo event? That's a separate event though and can be ignored for the network interception/logging purposes by clients who don't need it. In WebDriver BiDi the structured cookies will always be sent for all use cases 🤔  

I guess that's okay then (multiple network events in CDP is a big pain so we should avoid having a similar mechanism in BiDi) and some clients will likely need a snapshot of the stored cookies. If in the future there is a need for the ability to opt out from receiving the structured cookies to reduce amount of transferred data between the driver and the client, I guess that would be possible.

> that part was already in network request logging

Yeah, I overlooked that as I was reading the preview version and has not followed the spec closely back then. Thanks for the context!